SofaKernel/modules/SofaBaseMechanics/MechanicalObject.inl

```diff
@@ -1087,12 +1085,15 @@ void MechanicalObject<DataTypes>::init()
         numa_set_preferred(this->getContext()->getProcessor()/2);
 #endif
 
-    //Look at a topology associated to this instance of MechanicalObject by a tag
-    this->getContext()->get(m_topology, this->getTags());
-    // If no topology found, no association, then look to the nearest one
-    if(m_topology == NULL)
-        m_topology = this->getContext()->getMeshTopology();
+    if (!l_topology && d_useTopology.getValue())
+    {
+        l_topology.set( this->getContext()->getActiveMeshTopology() );
+    }
 
+    if (l_topology)
+    {
+        sout << "Initialization with topology " << l_topology->getTypeName() << " " << l_topology->getName() << " ( " << l_topology->getNbPoints() << " points)" << sendl; // << (l_topology->hasPos() ? "WITH" : "WITHOUT") << " positions)"
```

Comment:
In the sofa-consortium master's branch sout is internally using msg_info(), in a similar way serr is in fact using msg_warning(). So there is absolutely no interest in using them and anyone can safely replace sout <<  by msg_info() and serr << msg_warning() because it will not change the behavior (and it brings several benefits).
 
Having the messages routed to the component's log and/or general console is not decided at the components level but done by the "MessagesHandler". Currently in runSofa the messages handlers installed by default are printing all the messages in the console and add them to the component's log (which is used by the GUI to 'nicely' display the messages in the panels).